api-client/src/main/java/com/xing/api/data/messages/Conversation.java

```diff
@@ -26,135 +29,43 @@
 /**
  * Java representation of a Conversation.
  */
-public class Conversation implements Serializable {
+@AutoValue
+public abstract class Conversation implements Serializable {
```

Comment:
> Going back to the builder: If we're gonna create this objects manually I can add it, otherwise if it's gonna be only created by moshi (which is the case I think) I don't think is necessary.
In any case, we can add the builder later if we need it. So ignored for now.

> Is it really necessary? Think that in order to do it properly we would need 3 tests per object field: one with normal value -> no exception and two more for null-not present value -> exception. That means making 8*3 = 24 tests just to check that autovalue throws an exception in that case... I don't see the point.
The point is not to add tests for each case. The point is to make sure, that the api will always return non null values. Otherwise, the possible nulls should be annotated, otherwise we break the usage of the api.